bin/projects/dbatools/dbatools/Utility/Size.cs

```diff
@@ -179,34 +173,23 @@ public Size(long Byte)
             this.Byte = Byte;
         }
 
-        /// <summary>
-        /// Some more interface implementation. Used to sort the object
-        /// </summary>
-        /// <param name="obj">The object to compare to</param>
-        /// <returns>Something</returns>
+        /// <inheritdoc cref="IComparable{Size}.CompareTo"/>
+        /// <remarks>For sorting</remarks>
         public int CompareTo(Size obj)
         {
-            if (Byte == obj.Byte) { return 0; }
-            if (Byte < obj.Byte) { return -1; }
-
-            return 1;
+            return Byte.CompareTo(obj.Byte);
         }
 
-        /// <summary>
-        /// Some more interface implementation. Used to sort the object
-        /// </summary>
-        /// <param name="obj">The object to compare to</param>
-        /// <returns>Something</returns>
+        /// <inheritdoc cref="IComparable.CompareTo"/>
+        /// <remarks>For sorting</remarks>
+        /// <exception cref="ArgumentException">If you compare with something invalid.</exception>
         public int CompareTo(Object obj)
         {
-            try
+            if (obj is Size)
             {
-                if (Byte == ((Size)obj).Byte) { return 0; }
-                if (Byte < ((Size)obj).Byte) { return -1; }
-
-                return 1;
+                return CompareTo((Size) obj);
             }
-            catch { return 0; }
+            throw new ArgumentException(String.Format("Cannot compare a {0} to a {1}", typeof(Size).FullName, obj.GetType().FullName));
```

Comment:
I think its perfectly ok for this to throw an exception. If you disagree, create a test that shows this exception happening in any use case that's reasonable.